input/vlct/dual_energy_shock_tube/generate_shock_soln.py

```diff
@@ -0,0 +1,290 @@
+# this produces a tables of solutions to test against the Sod Shock
+# it's a little over-engineered but can be reused to prepare solutions to other
+# shock tube test problems
+
+# I have confirmed that the wavespeeds are consistent with the ones used in
+# Athena for the Sod Shock Tube test Problem
+
+import numpy as np
+
+def sound_speed(p,rho, gamma):
+    return np.sqrt(gamma*p/rho)
+
+class RarefactionFanSoln:
+    """ 
+    Solution to the Rarefaction fan. 
+
+    K is a variable that should be substituted for L or R. It is specified
+    by passing True or False to left
+    """
+    def __init__(self, gamma, rhoK, pK, uK, p_star, u_star, rho_starK,
+                 left = True):
+        self.gamma     = float(gamma)
+        self.rhoK      = float(rhoK)
+        self.pK        = float(pK)
+        self.uK        = float(uK)
+        self.p_star    = float(p_star)
+        self.u_star    = float(u_star)
+        self.rho_starK = float(rho_starK)
+        self.left = left
+
+        self.aK = sound_speed(self.pK,self.rhoK, self.gamma)
+        # eqns 4.54/4.61 from Toro
+        self.a_starK = self.aK * (self.p_star/self.pK)**(0.5 - 0.5/self.gamma)
+
+    def head_speed(self):
+        # eqns 4.55/4.62 from Toro
+        if self.left:
+            return self.uK - self.aK
+        else:
+            return self.uK + self.aK
+
+    def tail_speed(self):
+        # eqns 4.55/4.62 from Toro
```

Comment:
Potentially nitpicking, but it'd be great if you could add the Toro reference to the top (i.e., which edition you used).
I made some similar comments in a different code that confused some people who tried to understand the comment while looking at a different edition.

Reply:
Good point (I didn't realize there was that much variation between editions)